src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -197,10 +208,37 @@ impl<S: LocalStateStore> WriteFuture<S> {
         }
     }
 
+    fn paused(
+        duration: Duration,
+        stream: BoxedMessageStream,
+        write_state: LogStoreWriteState<S>,
+    ) -> Self {
+        Self::Paused {
+            sleep_future: Box::pin(sleep(duration)),
+            future: stream.into_future(),
+            write_state,
+        }
+    }
+
     async fn next_event(
         &mut self,
     ) -> StreamExecutorResult<(BoxedMessageStream, LogStoreWriteState<S>, WriteFutureEvent)> {
         match self {
+            WriteFuture::Paused {
+                sleep_future,
+                future,
+                ..
+            } => {
+                sleep_future.await;
+                let (opt, stream) = future.await;
```

Comment:
When we have passed the `sleep_future.await` and gets pending at `future.await`, if this future of `next_future` is dropped and recreated, we will still be at `WriteFuture::Paused`, and then we will poll the `sleep_future` again. However, the `sleep_future` have been ready, and the behavior of polling it is undefined, may be pending forever, or panic.

Reply:
Fixed, no longer poll the secondary future, thanks to @wenym1's idea below, where we can directly store the message, and replay it after pause.